### DISCUSSION TITLE
{project} {package} placeholders in CIBW_ENVIRONMENT

### DISCUSSION THREAD

**Discussion by sevagh** [pypa/cibuildwheel#1121] [General]
Hello,

I want to set an environment variable dynamically based on the project path, e.g.:
```
CIBW_ENVIRONMENT: "C_INCLUDE_PATH={project}/c/include"
CIBW_ENVIRONMENT_PASS_LINUX: C_INCLUDE_PATH
```

I can't use the setup.py include_dirs for this, it has to be an environment variable. Is this possible?

**Comment by sevagh** [comment]
Using `$(pwd)` instead worked.

**Reply by joerick** [reply]
yes, inside the build, `{project}` is always the same as `$(pwd)`.

p.s. you don't need to `CIBW_ENVIRONMENT_PASS_LINUX` a variable that you're defining in CIBW_ENVIRONMENT. `CIBW_ENVIRONMENT_PASS_LINUX` is for variables that are defined before invoking cibuildwheel (i.e. in the 'host' environment)

**Comment by hzhangxyz** [comment]
This does not works for windows, $(pwd) gives ""/d/a/project/project/" which cannot be recognized by windows native programs.

**Reply by hzhangxyz** [reply]
such as cmake
